provider/cloudflare/cloudflare_test.go

```diff
@@ -829,6 +829,25 @@ func TestCloudflareListZoneInternalErrors(t *testing.T) {
 	}
 }
 
+func TestCloudflareListZoneDetailsBadGatewayErrors(t *testing.T) {
+	// Create a mock client that returns a bad gateway error
+	client := NewMockCloudFlareClient()
+	client.listZonesContextError = &cloudflare.Error{
+		StatusCode: 502,
+		ErrorCodes: []int{20000},
```

Comment:
```suggestion
		ErrorCodes: []int{502},
```

Reply:
Done.. sorry missed that.. also the test above had the same problem changed it too